### PR TITLE
XDR get-alert enhancement

### PR DESCRIPTION
**Pull request by dansterenson**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/47493)

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 1]
```diff
@@ -3140,21 +3144,329 @@ script:
...
+      - ANONYMOUS
+      - ' APPLICATION'
+      - ' COMPUTE'
+      - ' FEDERATED_IDENTITY'
+      - ' SERVICE'
+      - ' SERVICE_ACCOUNT'
+      - ' TEMPORARY_CREDENTIALS'
+      - ' TOKEN'
+      - ' UNKNOWN'
+      - ' USER'
```
Comment (yaakovi) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:3191]: Are the spaces intentional?

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py [hunk 2]
```diff
@@ -2575,6 +2585,67 @@ def test_get_original_alerts_command(requests_mock):
     assert event.get('raw_log', {}).get('userIdentity', {}).get('accountId') == 'ID'  # assert vendor filter is correct
 
 
+@freeze_time("2022-05-03 11:00:00 GMT")
+def test_get_alert_by_filter(requests_mock, mocker):
```
Comment (yaakovi) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py:2589]: add a wrapper class for your tests to encapsulate them together:
```python
class TestGetAlertByFilter:
    @staticmethod
    def test_sanity():
        ...
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 3]
```diff
@@ -226,6 +296,73 @@ def clear_trailing_whitespace(res):
...
+    start_time = args.pop('start_time', None)
+    end_time = args.pop('end_time', None)
+
+    if (start_time or end_time) and ('time_frame' not in args):
+        raise DemistoException('Please choose "custom" under time_frame argument when using start_time and end_time '
+                               'arguments')
+
+    for arg_name, arg_value in args.items():
+        if arg_name not in valid_args:
+            raise DemistoException('Argument is not valid')
```
Comment (yaakovi) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:316]: state the invalid arg name

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 4]
```diff
@@ -177,6 +177,59 @@
...
 
+class AlertFilterArg:
+    def __init__(self, search_field: str, search_type: Optional[str], arg_type: str, option_mapper: dict = None):
+        self.search_field = search_field
+        self.search_type = search_type
+        self.arg_type = arg_type
+        self.option_mapper = option_mapper
+
+
+def init_filter_args_options():
```
Comment (yaakovi) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:188]: nice solution

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py [hunk 5]
```diff
@@ -47,6 +47,16 @@ def get_incident_extra_data_by_status(incident_id, alerts_limit):
 ''' TESTS FUNCTIONS '''
 
 
+# Note this test will fail when run locally (in pycharm/vscode) as it assumes the machine (docker image) has UTC timezone set
```
[the remark this answers is not in the document]
Comment (dansterenson) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py:50]: Passes on `demisto-sdk lint` but fails when executed from pycharm.

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml (hunk 1 above)
Comment (dansterenson) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:3191]: Nope, fixed

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py (hunk 2 above)
Comment (dansterenson) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py:2589]: Done

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py (hunk 3 above)
Comment (dansterenson) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:316]: Done

**Comment by dansterenson** [comment]
@yaakovi There are few merge conflicts and they will be fixed after the review.

**Review by yaakovi — APPROVED** [review]

**Pull request merged**